### PR TITLE
Move string equals to Rust.

### PR DESCRIPTION
**Pull request by shaleh**

### CODE REVIEW

**Comment by brotzeit** [comment]
I'd really like to approve more of your PRs, but they are mostly not easy to review(at least for me). Please make smaller PRs so it's easier to understand the changes. I don't want to approve PRs that I don't understand completely.

**Comment by shaleh** [comment]
@brotzeit, this is the simplest version.

**Comment by brotzeit** [comment]
@shaleh Thanks.

File: rust_src/src/fns.rs [hunk 1]
```diff
@@ -264,4 +264,20 @@ pub fn concat(args: &mut [LispObject]) -> LispObject {
...
+    kind: equal_kind::Type,
+) -> bool {
+    let s1 = o1.as_string_or_error();
+    let s2 = o2.as_string_or_error();
+
+    s1.len_chars() == s2.len_chars()
+        && s1.len_bytes() == s2.len_bytes()
+        && s1.as_slice() == s2.as_slice()
+        && (kind != equal_kind::EQUAL_INCLUDING_PROPERTIES
+            || unsafe { compare_string_intervals(o1, o2) })
```
[the remark this answers is not in the document]
Comment (shaleh) [reply rust_src/src/fns.rs]: No, I applied deMorgan's.

`foo != bar && a != b` is the same as `!(foo == bar || a ==b)`

The original code used a series of `!=` and then ended in `== or !compare`.

[the remark this answers is not in the document]
Comment (brotzeit) [reply rust_src/src/fns.rs]: Oh, cool. Sounds familiar =)

**Pull request merged**